hs/src/Reach/UnrollLoops.hs

```diff
@@ -119,15 +113,15 @@ instance Unroll DLStmt where
     DL_Set at v a -> return $ DL_Set at v a
     DL_LocalIf at c t f -> DL_LocalIf at c <$> ul t <*> ul f
     DL_LocalSwitch at ov csm -> DL_LocalSwitch at ov <$> ul csm
-    DL_ArrayMap at ans x a i fb -> do
-      (_, x') <- ul_explode at x
-      r' <- zipWithM (\xa iv -> fu_ fb [(a, xa), (i, (DLA_Literal $ DLL_Int at iv))]) x' [0..]
+    DL_ArrayMap at ans xs as i fb -> do
+      (_, xs') <- unzip <$> mapM (ul_explode at) xs
+      r' <- zipWithM (\xa iv -> fu_ fb $ (zip as xa) <> [(i, (DLA_Literal $ DLL_Int at iv))]) (transpose xs') [0..]
```

Comment:
That's what `fu_` does ... "freshen up"... that's our name for alpha-varying. Basically because we're unrolling, we need to make new variables for the body of the loop because it will occur many times now